### PR TITLE
Parent/Secondary Wayfinding

### LINKED ISSUE #337: Parent / Secondary docs wayfinding
Currently YAML supports displaying an article in two locations via the `parent` and `secondary` yaml attributes. Currently when viewing these docs the `parent` will always take the active state. Lets discuss a resolution.

See Popovers as an example. It appears in both Content Areas and Editing section.

http://staging.design.rax.io/design-system/components/popovers.html

### PR DESCRIPTION
**Pull request by nimbinatus**
Fixes #337

### CODE REVIEW

File: templates/_includes/nav.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
         </header>
         {%- for famItem in site.data.families -%}
           {%- assign currentFam = famItem.parent | slugify -%}
-          <a href="{{site.url}}/components/{{currentFam}}.html" {%- if currenturl contains currentFam -%} class="current" {%- elsif page.parent contains famItem.parent -%} class="current"{%- elsif itemparent contains famItem.parent -%} class="current"{%- endif -%}>{{famItem.display}}</a>
+          <a href="{{site.url}}/components/{{currentFam}}.html" {%- if currenturl contains currentFam -%} class="current"{%- elsif page.parent contains famItem.parent -%} class="current"{%- elsif itemparent contains famItem.parent -%} class="current"{%- endif -%} id="{{currentFam}}">{{famItem.display}}</a>
```
Comment (technabors) [inline templates/_includes/nav.html]: Working as intended to close out the issue but I notice when the secondary item is the `famItem.parent` it applies the class to the primary page and then quickly moves to the secondary item. Switching the order of the if statement should eliminate this quick shift. Try the below order:

Edit: Nevermind tried my suggestion locally and it didn't fix the issue. Any thoughts on how we can eliminate that quick shift?

Comment (nimbinatus) [reply templates/_includes/nav.html]: It applies the class because we need to keep that functionality for folks who might use bookmarks (i.e., users that haven't navigated there directly). The removal is triggered by Javascript, which fires after the page loads.

Comment (technabors) [reply templates/_includes/nav.html]: discussion occurred outside GH — 

Evan Nabors [2:51 PM] 
All looks good to me but I do notice that it quickly jumps from the primary to the secondary when you path from the secondary item. I.e. click selectors -> selector-strip and watch the nav. Not sure how we could eliminate that though. I tried reordering the if statement and that didn’t seem to work. :thinking_face:


[2:51] 
@santamaria ^


Laura Santamaria [2:52 PM] 
Yep, just responded to the comment on the PR


[2:52] 
essentially, that quick switch is there just in case someone uses bookmarks to navigate to the site


[2:53] 
so, if someone uses a bookmark to get to the component detail page, there’s an active nav listed. The active class is removed by javascript if the visitor navigates there from a different family page, so it has to fire after the page loads, which causes the quick switch


[2:53] 
at the moment, I couldn’t think of a way to prevent that quick switch while honoring folks that arrive via bookmark (edited)


[2:54] 
so, in short, I ensured there was a default, which is generated by Jekyll. Then, if the user had a different navigation, the JS fires to change it to match the user’s path to the site


Evan Nabors [2:54 PM] 
totes. Ok I’ll say approved in that case.


new messages
Laura Santamaria [2:55 PM] 
awesome sauce :smile:

**Comment by araiford** [comment]
I haven't dug into this code, and I'm a little unsure what this PR does.  Can you provide a high-level?

**Comment by nimbinatus** [comment]
@araiford sure. On the current site, the breadcrumbs and highlighted side nav item for a component details page shows the family listed as `parent` in the metadata. However, visitors can also get to a component details page from a family page listed as `secondary` in the metadata, so the breadcrumbs and side nav do not accurately represent how a user navigates to a component details page 100% of the time. This PR adds a JS file (`wayfinding.js`) that uses the browser's history to identify which family page the user came from to populate the breadcrumbs and the side nav active element. If a user browses to the component details page from a bookmark or some other page (e.g., the status page), the breadcrumbs and the side nav active element defaults to the `parent` family.

The changes to the other files include adding `id` tags that the JS can hook into and adding a bit of data to the `families` dataset to make the generation of the templates easier.

**Comment by technabors** [comment]
@araiford More discussion on this PR can be found on the following issue #337

**Comment by araiford** [comment]
@nimbinatus @technabors 

LGTM :shipit:
